packages/core/src/fetchers/fetchAllEntities.ts

```diff
@@ -0,0 +1,160 @@
+import { Asset, ContentfulClientApi, Entry } from 'contentful';
+import { MinimalEntryCollection } from './gatherAutoFetchedReferentsFromIncludes';
+import { uniqBy } from 'lodash-es';
+
+const MIN_FETCH_LIMIT = 1;
+export const fetchAllEntries = async ({
+  client,
+  ids,
+  locale,
+  skip = 0,
+  limit = 100,
+  responseItems = [],
+  responseIncludes = { Entry: [], Asset: [] },
+}: {
+  client: ContentfulClientApi<undefined>;
+  ids: string[];
+  locale: string;
+  skip?: number;
+  limit?: number;
+  responseItems?: Entry[];
+  responseIncludes?: MinimalEntryCollection['includes'];
+}) => {
+  try {
+    if (!client) {
+      throw new Error(
+        'Failed to fetch experience entities. Required "client" parameter was not provided',
+      );
+    }
+
+    if (!ids.length) {
+      return {
+        items: [],
+        includes: {
+          Entry: [],
+          Asset: [],
+        },
+      };
+    }
+
+    const query = { 'sys.id[in]': ids, locale, limit, skip };
+
+    const {
+      items,
+      includes,
+      total: responseTotal,
+    } = await client.withoutLinkResolution.getEntries({ ...query });
+
+    responseItems.push(...(items as Entry[]));
+    responseIncludes?.Entry?.push(...(includes?.Entry || []));
+    responseIncludes?.Asset?.push(...(includes?.Asset || []));
+
+    if (skip + limit < responseTotal) {
```

Comment:
Maybe this is highly subjective, and maybe I am missing some obvious logic. But to me `skip + limit` sounds like `take age of a tree + multiply by the average speed of hamster running away from a snake and divide by latitude of Sydney` ... 

Again maybe it's just me, but `alreadyFetchedEntities + lastPageEntityCount > totalEntityCount` sounds more natural and intuitive 🤷